Templates/ScriptCanvasNode/Template/README.md

```diff
@@ -31,6 +31,10 @@ scripts\o3de register -esgp Gems\${GemName} -es Gems\${GemName}\Code\Source\${Na
 
 Configure & Generate your projects
 
+To your YourGem Code\CMakeList add your MyScriptCanvasNodes directory by appending
+add_subdirectory(Source/MyScriptCanvasNodes)
```

Comment:
This step is not needed.

Once the folder container containing the ScriptCanvas Node directory is registered with the Gem in it's gem.json file via the "external_subdirectories" key, then add_subdirectory will be called on it automatically via the subdirectory visitation logic in [Subdirectories.cmake](https://github.com/o3de/o3de/blob/development/cmake/Subdirectories.cmake#L327-L356) when that gem is active.

The Atom gem has an [example](https://github.com/o3de/o3de/blob/development/Gems/Atom/gem.json#L19-L32) of visiting sub-gems and regular external_subdirectories.
A "regular" external_subdirectory is just a directory with a CMakeLists.txt in it, but no `gem.json`. 
A "Gem" is an external_subdirectory with a `gem.json` in it.

Any external_subdirectories registered with the current project, the engine being used or any gem's being used by the current project will have recursively have all of it's children "external_subdirectories" entries gathered and then `add_subdirectory` invoked on each path.

Reply:
You're right, I think it was carryover from the first iteration that I missed cleaning. Thanks for the update!

Reply:
@lumberyard-employee-dm the external sub directories apparently doesn't pass on the gem_name variable, at least it produces build errors for me "The target .Private.Object was not found, if the gem was not created using"

Which is unfortunate since therefor we need to pass the -r ${GemName} YourGem to the create-from-template
While with the manual addition of the add_subdirectory this is working.
Since the bit that gets replaced has the format ${something} it is prone to user error due to shell substitution. 

I have some further changes locally which I will push w/ further feedback from you and the amended commit to satisfy the Certificate of Origin.

Reply:
In that case if the gem name is needed it can be queried from the nearest ancestor gem.json using the [o3de_find_ancestor_gem_root](https://github.com/o3de/o3de/blob/development/cmake/Gems.cmake#L18-L23) method.

That can be added to the template CMakeLists.txt to query the gem name and therefore the need to specify `-r ${GemName} <YourGem>` argument isn't needed.

Reply:
Try adding this code to the top of the CMakeLists.txt in the template
```
# Query the gem name from the gem.json file if possible
# otherwise fallback to using ${GemName}
o3de_find_ancestor_gem_root(gem_path gem_name "${CMAKE_CURRENT_SOURCE_DIR}")
if (NOT gem_name)
    set(gem_name "${GemName}")
endif()
```